gluten-data/src/main/scala/io/glutenproject/execution/GlutenHashAggregateExecTransformer.scala

```diff
@@ -420,7 +422,7 @@ case class GlutenHashAggregateExecTransformer(
 
     if (extractStructNeeded()) {
       aggParams.extractionNeeded = true
-      aggRel = applyExtractStruct(context, aggRel, operatorId, validation)
+      aggRel = applyExtractStruct(context, aggRel, operatorId, validation, originalInputAttributes.size)
```

Comment:
We should be noted `originalInputAttributes = child.output` and `originalInputAttributes.size` is the input size of AggRel. But we need the output size of AggRel here. It's would be better if we can double-check all the used emit indices.